### PR TITLE
P2 signup: add P2Details step

### PR DESCRIPTION
**Pull request by lamosty**
In this PR, we add the "second" step in the P2 signup flow at wordpress.com/start/p2:

## Testing instructions

Please test all the possible variations:

- log in with a wpcom account
- register a new account
- click on register a new account but then click to "Already have a wpcom account?"
- try a completely different flow like /start/free if it still works properly

### CODE REVIEW

**Comment by lamosty** [comment]
On hold until https://github.com/Automattic/wp-calypso/issues/43575 is addressed.

**Comment by lamosty** [comment]
This should be no longer blocked after introducing https://github.com/Automattic/wp-calypso/pull/44172. Should we continue working on this or close it for now? @lezama @evilluendas

**Comment by evilluendas** [comment]
Great news! I think we should still add this step; in a couple of informal user testing sessions I've noticed that it's easy to miss the log in link on the Create your account step of the standard wpcom flow.

image

This step makes it more explicit and easier to move forward in the flow.

Could you let me know once it's ready to test and tweak so I can make the last polishing of the styling? 🙌

**Comment by lamosty** [comment]
After a lot of hacking the flow is ready for a review!

Please test all the possible variations:
- log in with a wpcom account
- register a new account
- click on register a new account but then click to "Already have a wpcom account?"
- try a completely different flow like `/start/free` if it still works properly

**Comment by lamosty** [comment]
@ramonjd Hi. I added you for a review since you've helped us with Calypso signup flows before :) Could you please check whether we are not breaking the main wpcom signup flows here? I don't think so but just to be sure.

File: client/signup/steps/p2-login-signup/index.jsx
Comment (lezama) [inline client/signup/steps/p2-login-signup/index.jsx:97]: `translate`

File: client/signup/steps/p2-login-signup/index.jsx
Comment (lezama) [inline client/signup/steps/p2-login-signup/index.jsx:108]: `translate`

**Comment by evilluendas** [comment]
Works great! I just pushed a tiny change to a couple of margins, but the rest looks on point. Fantastic work overall! 👌 

There's one small bug I've noticed: if you go back to change the details, the next time you see this step the buttons don't have space between. I can reproduce most of the time, but not always (which is weird). Here's a gif:



Finally, the translations are offered again when you return to the first step to change something, which can be a bit annoying as it pushes the form downwards with no warning. Would it be feasible to hide them once you've discarded it for the first time? It's not a deal breaker in any case.

**Comment by evilluendas** [comment]
Another small nitpick I just noticed: instead of `start/p2/p2-login-signup` could the url of this step be `start/p2/p2-details`? Or even better, `start/p2/details`? (I seem to recall that removing the `p2-` part isn't recommended, but I don't remember clearly).

**Comment by lamosty** [comment]
> Another small nitpick I just noticed: instead of start/p2/p2-login-signup could the url of this step be start/p2/p2-details? Or even better, start/p2/details? (I seem to recall that removing the p2- part isn't recommended, but I don't remember clearly).

I can rename it to `P2Details` so it would be `p2-details` in the url but I wouldn't drop the "p2" part as it's easier to grok code-wise.

**Comment by lamosty** [comment]
> One thing that I noticed if I am logged out and do Sign up with WordPress and then Already have an account it just gets stuck in the Hooray screen, refreshing takes me back to P2 Details page. https://d.pr/v/9gswmm

Ah, yep, it's being addressed here https://github.com/Automattic/wp-calypso/pull/44400 and then a bit more in this PR once that one lands and I'll get to rebase it. Sorry for not putting `in progress` label.

**Comment by lamosty** [comment]
After landing #44400, looks like all the flows are working!

@unDemian could you give it another review, please? I feel we are close to merging!

**Review by unDemian — APPROVED** [review]
Works as described! :shipit:

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4245958

Thank you @lamosty for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4245958).

**Pull request merged**
